### PR TITLE
net/dns: call LUID.SetDNS instead of writing to registry directly

### PR DESCRIPTION
**Pull request by maisem**
This calls the approrpiate Win32 API to do the same.

Updates #1659

Signed-off-by: Maisem Ali <maisem@tailscale.com>

### CODE REVIEW

**Comment by bradfitz** [comment]
Typo of "appropriate" in commit message/PR text.

File: net/dns/manager_windows.go [hunk 1]
```diff
@@ -134,53 +143,35 @@ func (m windowsManager) setPrimaryDNS(resolvers []netip.Addr, domains []dnsname.
...
-			return err
-		}
-	} else if err := key4.SetStringValue("SearchList", strings.Join(domStrs, ",")); err != nil {
+	// Disable LLMNR on the Tailscale interface. We don't do
+	// multicast, and we certainly don't do LLMNR, so it's pointless
+	// to make Windows try it.
+	//
+	// TODO(maisem): The same effect can be achieved by setting
+	// winipcfg.DnsInterfaceSettings.EnableLLMNR to 0. Currently upstream does
+	// not support setting the DnsInterfaceSettingsFlagEnableLLMNR flag, use it
```
Comment (danderson) [inline net/dns/manager_windows.go:160]: Is there an upstream bug/PR somewhere? Link?

File: net/dns/manager_windows.go [hunk 2]
```diff
@@ -118,14 +118,23 @@ func (m windowsManager) setSplitDNS(resolvers []netip.Addr, domains []dnsname.FQ
...
+	tsLUID, err := winipcfg.LUIDFromGUID(&tsGUID)
+	if err != nil {
+		return err
+	}
+	ips4 := make([]net.IP, 0, len(resolvers))
+	ips6 := make([]net.IP, 0, len(resolvers))
 	for _, ip := range resolvers {
 		if ip.Is4() {
-			ipsv4 = append(ipsv4, ip.String())
+			b := ip.As4()
```
Comment (danderson) [inline net/dns/manager_windows.go:133]: If we're upstreaming changes to winipcfg, maybe also propose one to use net/netip types?

**Review by danderson — APPROVED** [review]

**Comment by dblohm7** [comment]
@maisem Could you please clarify what the impetus is for doing this? Is the concern that using the API might allow the changes to take effect more quickly than directly editing the registry?

File: net/dns/manager_windows.go (hunk 2 above)
Comment (maisem) [reply net/dns/manager_windows.go:133]: It already does - https://github.com/tailscale/tailscale/pull/5343

File: net/dns/manager_windows.go (hunk 1 above)
Comment (maisem) [reply net/dns/manager_windows.go:160]: Not yet, will file.

**Comment by maisem** [comment]
this was a shot in the dark when trying to figure out windows dns stuff, don't think its necessary any more.

**Pull request closed without merging**